src/platform/plugins/shared/discover/public/application/main/components/top_nav/use_top_nav_links.tsx

```diff
@@ -95,7 +102,8 @@ export const useTopNavLinks = ({
       if (
         services.triggersActionsUi &&
         services.capabilities.management?.insightsAndAlerting?.triggersActions &&
-        !defaultMenu?.alertsItem?.disabled
+        !defaultMenu?.alertsItem?.disabled &&
+        discoverParams.authorizedRuleTypeIds.includes(ES_QUERY_ID)
```

Comment:
Nice! So this change would also close another bug, right?
- https://github.com/elastic/kibana/issues/184564